sync/src/lib.rs

```diff
@@ -153,8 +153,14 @@ impl NetworkProtocolHandler<SyncMessage> for EthSync {
 	}
 
 	fn message(&self, io: &NetworkContext<SyncMessage>, message: &SyncMessage) {
-		if let SyncMessage::BlockVerified = *message {
-			self.sync.write().unwrap().chain_blocks_verified(&mut NetSyncIo::new(io, self.chain.deref()));
+		match *message {
+			SyncMessage::BlockVerified => {
```

Comment:
`sync` should not care about `BlockVerified` event at all. Please merge `chain_blocks_verified` function into `chain_new_blocks`. These two should be called for the same event and `chain_blocks_verified` is an incorrect name.